documentation/src/main/asciidoc/topics/protostream.adoc

```diff
@@ -0,0 +1,163 @@
+link:https://developers.google.com/protocol-buffers/docs/overview[Developer Guide] for an in-depth
+explanation of the encoding format and best practices.
+
+Protocol Buffers, Protobuf for short, provide a platform independent encoding format that utilises .proto schema files to
+provide a structured representation of entities that can be easily evolved over time whilst maintaining backwards compatibility.
+
+Protobuf is all about structured data, so the first thing todo is to define the structure of your data. This is
+accomplished by declaring Protobug message types in .proto files, as shown in the example below.
+
+[[book_sample_proto]]
+[source,proto]
+.library.proto
+----
+include::config_examples/library.proto[]
+----
+
+In this example .proto file, we define an entity (message type in Protobuf speak) named _Book_, which is contained in the
+package _book_sample_. The _Book_ entity declares several fields of primitive types and a repeatable field (an
+array basically) named _authors_, which is the _Author_ message type also declared in _library.proto_.
+
+There are a few important notes we need to make about Protobuf messages:
+
+* Nesting of messages is possible, but the resulting structure is strictly a tree, never a graph
+* There is no concept of type inheritance
+* Collections are not supported but arrays can be easily emulated using repeated fields
+
+
+=== Marshallers
+As described in the <<Protobuf Schemas,previous section>>, a fundamental concept of the Protobuf format is the defining
+of messages in the .proto schema to determine how an entity is represented. However, in order for our Java
+applications to utilise the Protobuf format to transmit/store data, it's necessary for our Java objects to be encoded.
+This is handled by the ProtoStream library and it's configured _Marshaller_ implemenations, which convert plain old
+Java objects into the Protobuf format.
+
+=== SerializationContext
+A fundamental component of the ProtoStream library is the SerializationContext. This is a repository for Protobuf
+type definitions, loaded from .proto files, and their accompanying Marshaller implementations. All ProtoStream marshalling
+operations happen in the context of a provided SerializationContext.
+
+[[protostream_usage]]
+== Usage
+ProtoStream is able to handle the following types, as well as their unboxed equivalent in the case of primitive types,
+without any additional configuration:
+
+* String
+* Integer
+* Long
+* Double
+* Float
+* Boolean
+* byte[]
+* Byte
+* Short
+* Character
+* java.util.Date
+* java.time.Instant
+
+Support for additional Java objects is possible by configuring the <<SerializationContext>> using <<SerializationContextInitializer>>s.
+
+{brandname} directly integrates with the ProtoStream library by allowing users to configure implementations of the ProtoStream
+SerializationContextInitializer interface. These implementations are then used to initialise the various SerializationContext's used by
```

Comment:
The pluralization of SerializationContext should not have an apostrophe, but then we cause ambiguity with the identifier name.
@oraNod should we use monospace for identifiers and standard formatting for the 's' pluralization to indicate it isn't part of the identifier ?

Reply:
I think that would be the way to go, `SerializationContext`s, even though it would look a bit odd when rendered. my preference would be something like:

initialise the various `SerializationContext` objects/constructs/configurations used by 

(not sure exactly which word is precise, but whatever the SerializationContext interface provides)

Reply:
I think instances is what we need.